### DISCUSSION TITLE
Nozzle x, y, Offset (Not Probe X,Y)?

### DISCUSSION THREAD

**Discussion by kentsangcanada** [Jyers/Marlin#235] [Ideas]
I upgraded my Hotend so my Nozzle is not at the original MK8 position
is there anyway in the next release that includes a Nozzle Offset in advanced?

edit: I mean Nozzle X, Y offset, say when you upgrade to a V6 or other direct extruder hotend
edit: Nozzle X,Y offset (Not Probe X,Y)

**Comment by Schnabulation** [comment]
...go to "Prepare" - there is a Z-offset menu where you can set the offset.

**Reply by kentsangcanada** [reply]
I forgot to write Nozzle x, y offset
I know about the Z-offset, but you upgrade the hot end to something like a V6 or other extruder hotends there is going to be an x,y offset

**Reply by Schnabulation** [reply]
This is also already implemented - you can find these settings under "Control" -> "Advanced". There you can set the X, and Y offset.

**Reply by kentsangcanada** [reply]
sorry again, currently in control>advanced> ... thats PROBE offset from the Nozzle, but when you upgrade to V6 your X,Y for the NOZZLE itself is MOVED away from the original XY of the OEM MK8 hotend

I think there is a GCODE for moving the XY of the nozzle, but it would probably work better if its initiated in the firmware, then start the UBL mesh grid. 

Because the UBL/MESH bed uses original Creality nozzle XY, THEN offset the PROBE by the "probe XY" to calcuated the mesh

**Reply by Schnabulation** [reply]
Ok, I understand now...

...may I suggest you look into compiling your own firmware for this quite specific need?

**Reply by kentsangcanada** [reply]
I may have to do that, but I am sure Lots of people in the community doing HeroMe Gen5 or other Direct extruder mods, and Jysers' Firmware is HOT GOTO firmware right now, I though it would be a great IDEA to put into Control>Advanced

Since I notice Jyser implemented MUCH higher estep now (I though nozzle offset is a great idea), cuz by default BMG etruder requires 415estep, and default creality is only 93, and 2 version of the earlier firmware had like a 255 max estep... yes i know you can init estep with M92 E415, then M500. but its amazing that he added it to his firmware. 

Thank you for all your hard work
